src/ConfigSyntax.cpp

```diff
@@ -3,7 +3,135 @@
 #include <Logger.hpp>
 #include <Token.hpp>
 
-void ConfigParser::syntaxCheck(std::vector<Token>::iterator token)
+#include <stdexcept>
+#include <vector>
+
+void printLine(std::vector<Token>::iterator it)
+{
+	Logger &logger = Logger::getInstance();
+
+	std::string line;
+	while (it->getType() == TokenType::WORD)
+	{
+		line += " " + it->getString();
+		it++;
+	}
+	logger.log(DEBUG, line);
+}
+
+// Finds the end of the block. expects to be called at block_identifier token.
+std::vector<Token>::iterator findBlockEnd(std::vector<Token> tokenlist,
+										  std::vector<Token>::iterator it)
+{
+	int stack;
+	Token &start_block = *it;
+
+	it++;
+	if (start_block.getString() == "location")
```

Comment:
What happens when "location != location"?

Reply:
Being a location_block it always has a request_target after the block_identifier. Meaning it should skip it. 

The only case where it could possibly enter this function otherwise is in a server_block. cause otherwise it will be filtered out as an unidentified_block or an invalid KEY (see 
void ServerSettings::addValueToServerSettings
&&
LocationSettings::LocationSettings(Token) constructor